### PR TITLE
Fix grep expression when parsing for namespace

### LINKED ISSUE #53: Invalid KF_DEPLOYMENT_NAMESPACE from env grep
**Describe the bug**
When launching a notebook using the v0.2.0 or v0.2.1 image, we get an error due to a bad value in `KF_DEPLOYMENT_NAMESPACE`. This is due to the fact that both `JUPYTER_IMAGE` and `JUPYTER_IMAGE_SPEC` are defined in the env, so the grep/cut [here](https://github.com/opendatahub-io/s2i-lab-elyra/blob/f811bf1e3056bb4b011eb3a64c7d59e07f14700c/setup-elyra.sh#L17) returns two matching results with a newline between them.
Error looks like this (actual image name replaced with <imagename> for clarity):
`Validation failed for instance 'my_kfp' using the kfp schema with error: 'http://ml-pipeline.<imagename>\n<imagename>/pipeline' is not a 'uri'.`

**To Reproduce**
1. Launch a new Elyra notebook from v0.2.0 or v0.2.1 image on a Jupyterhub instance that defines both JUPYTER_IMAGE and JUPYTER_IMAGE_SPEC env vars.
2. This will fail. Logs can be inspected to see the error.

**Expected behavior**
A single line is used for the `KF_DEPLOYMENT_NAMESPACE`. This could be achieved in several ways. I believe `JUPYTER_IMAGE` and `JUPYTER_IMAGE_SPEC` should always match, so the grep could simply have the `-m1` param added so that it only returns the first result.

**Screenshots**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by akchinSTC**
Signed-off-by: Alan Chin <akchin@us.ibm.com>

## Related Issues and Dependencies

Fixes #53 

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

Updates the regex to look only for the JUPYTER_IMAGE env var when looking up the namespace.

### CODE REVIEW

File: setup-elyra.sh [hunk 1]
```diff
@@ -14,7 +14,7 @@ export COS_BUCKET=${COS_BUCKET:-default}
 # If Kubeflow credentials are not supplied, use default Kubeflow installation credentials
 KF_USERNAME="${KF_USERNAME:=user@example.com}"
 KF_PASSWORD="${KF_PASSWORD:=12341234}"
-KF_DEPLOYMENT_NAMESPACE=$(env | grep JUPYTER_IMAGE | cut -d'/' -f2)
+KF_DEPLOYMENT_NAMESPACE=$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
```
Comment (MightyNerdEric) [inline setup-elyra.sh:17]: This command does not seem to work in our OpenShift-based environment. I'm running the command on a working v0.1.3 image (our current fallback; we went straight from 0.1.3 to 0.2.1 before running into the issue here).

**Review by MightyNerdEric — CHANGES_REQUESTED** [review]

File: setup-elyra.sh (hunk 1 above)
Comment (akchinSTC) [reply setup-elyra.sh:17]: hrmm, 
my current test env consists of a openshift cluster 4.10 with the 1.2.0 operator and 1.4.0 manifests.
Could you let me know what env you're running on so I can narrow down the issue?

Comment (akchinSTC) [reply setup-elyra.sh:17]: include any configuration changes like customized manifests too, thanks!

Comment (ptitzler) [reply setup-elyra.sh:17]: Is it possible that in your environment the service account is limiting which secrets are exposed to the pod? The info in https://docs.openshift.com/container-platform/4.11/authentication/using-service-accounts-in-applications.html section "About automatically-generated service account token secrets" might also provide a clue.  Perhaps the output of `oc describe sa <service-account>` too.

**Comment by harshad16** [comment]
With Elyra installed on the ODH workbench images directly.
Please refer ODH base notebook images, to utilize elyra.

Closing this issue. Feel free to open the issue on 
Reference: https://github.com/opendatahub-io/notebooks

**Pull request closed without merging**